test: add validation tests for Elasticsearch tool operations

nehaaprasad opened the pull request:
CLOSES: #229975

## Summary

This PR addresses the review comments from #229497 by improving the Elasticsearch tool implementation. It extracts the operation validation logic into a well-named helper function `isOperationAllowed` and adds comprehensive API tests to ensure only certain operations are allowed through the AI assistant.

nehaaprasad committed:
added the new test file to the test suite by including

nehaaprasad committed:
Added the isOperationAllowed helper function

nehaaprasad committed:
This is the new API test file that covers various scenarios for the Elasticsearch tool restrictions

viduni94 commented on x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/ai_assistant/elasticsearch_tool/elasticsearch_tool.spec.ts:
```diff
@@ -0,0 +1,426 @@
+import expect from '@kbn/expect';
```
The existing tests related to this function is located in `x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/ai_assistant/complete/functions/elasticsearch.spec.ts`
Would you be able to move the new tests there?

viduni94 commented on x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/ai_assistant/elasticsearch_tool/elasticsearch_tool.spec.ts:
```diff
@@ -0,0 +1,426 @@
+        });
+
+        await observabilityAIAssistantAPIClient.editor({
+          endpoint: 'POST /internal/observability_ai_assistant/chat',
```
Instead of calling the `/chat` endpoint, you could use the existing helper function to invoke a chat complete request with the function call. Would you be able to update all the tests to use the `invokeChatCompleteWithFunctionRequest` helper method?

Example:
```
const responseBody = await invokeChatCompleteWithFunctionRequest({
        connectorId,
        observabilityAIAssistantAPIClient,
        functionCall: {
          name: ELASTICSEARCH_FUNCTION_NAME,
          trigger: MessageRole.User,
          arguments: JSON.stringify({
            method: 'POST',
            path: 'traces*/_search',
            body: {
              size: 0,
              aggs: {
                services: {
                  terms: {
                    field: 'service.name',
                  },
                },
              },
            },
          }),
        },
      });
```

viduni94 commented on x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/ai_assistant/elasticsearch_tool/elasticsearch_tool.spec.ts:
```diff
@@ -0,0 +1,426 @@
+        expect(requestBody.messages[0].content).to.eql(SYSTEM_MESSAGE);
+      });
+
+      it('should allow POST requests to _search endpoint', async () => {
```
```suggestion
      it('should allow POST requests to the _search endpoint', async () => {
```

viduni94 commented on x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/ai_assistant/elasticsearch_tool/elasticsearch_tool.spec.ts:
```diff
@@ -0,0 +1,426 @@
+      });
+    });
+
+    describe('Disallowed Operations', () => {
```
For all tests in this block, would you be able to check whether the function call response has an error that states the request is not allowed?

viduni94 commented on x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/ai_assistant/elasticsearch_tool/elasticsearch_tool.spec.ts:
```diff
@@ -0,0 +1,426 @@
+      });
+    });
+
+    describe('Edge Cases', () => {
```
```suggestion
    describe('Request paths with query parameters', () => {
```

viduni94 commented on x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/ai_assistant/elasticsearch_tool/elasticsearch_tool.spec.ts:
```diff
@@ -0,0 +1,426 @@
+    },
+  ];
+
+  describe('/internal/observability_ai_assistant/chat - Elasticsearch Tool Restrictions', function () {
```
After these tests are moved to the `tool: elasticsearch` test suite, the describe block can be renamed.

```suggestion
  describe('tool restrictions', function () {
```

viduni94 reviewed (commented):
Thank you for the PR @naaa760 
I've added some comments

nehaaprasad committed:
fix: update elasticsearch tool tests to properly validate function restrictions

- Replace system message validation with actual function execution testing
- Use invokeChatCompleteWithFunctionRequest helper as suggested by reviewer
- Test that allowed operations (GET, POST to _search) return success responses
- Test that disallowed operations (PUT, DELETE, PATCH, POST to non-search) return proper error messages
- Verify actual security restrictions rather than just system configuration
- Addresses reviewer feedback about validating function call responses vs system messages

naaa760 commented:
@viduni94 
Hi! I've updated the tests to use invokeChatCompleteWithFunctionRequest and now properly validate whether function calls succeed/fail with the right error messages instead of just checking system messages.

nehaaprasad committed:
Merge branch 'main' into test/validate-es-operations
